src/main/java/org/mqttbee/mqtt/handler/connect/MqttConnectHandler.java

```diff
@@ -82,11 +83,23 @@ public MqttConnectHandler(
 
     @Override
     public void channelActive(final ChannelHandlerContext ctx) {
-        addClientData(ctx.channel());
-        writeConnect(ctx);
+        if (!connectSent) {
+            addClientData(ctx.channel());
+            writeConnect(ctx);
+        }
         ctx.fireChannelActive();
     }
 
+    @Override
+    public void handlerAdded(ChannelHandlerContext ctx) {
+        super.handlerAdded(ctx);
```

Comment:
no-op and should be removed

Reply:
Why no-op? super does

`@Override
    public void handlerAdded(final ChannelHandlerContext ctx) {
        this.ctx = ctx;
    }
`

Reply:
Sorry, my fault, thought that it directly extends ChannelInboundHandlerAdapter